### PR TITLE
Refactor Network component

### PR DESCRIPTION
**Pull request by whymarrh**
This PR refactors the `Network` component, removing the nested function with the switch statement and the redundant variables.

### CODE REVIEW

File: ui/app/components/app/network.js [hunk 1]
```diff
@@ -21,160 +54,112 @@ export default class Network extends Component {
...
+      onClick,
+    } = this.props
+
+    const {
+      t,
+    } = this.context
+
     let providerName, providerNick, providerUrl
-    try {
+    if (this.props.provider) {
```
Comment (rickycodes) [inline ui/app/components/app/network.js]: nit: would it make sense to destructure `provider` from `this.props` like what you've done with `disabled`?

**Review by rickycodes — DISMISSED** [review]
LGTM from what I can tell

**Review by Gudahtt — DISMISSED** [review]
Thanks, looks good! Great to get rid of that odd IIFE

File: ui/app/components/app/network.js (hunk 1 above)
Comment (whymarrh) [reply ui/app/components/app/network.js]: Good point, I'll clean this up

**Comment by whymarrh** [comment]
@rickycodes @Gudahtt updated

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**